packages/flutter_tools/templates/plugin/ios.tmpl/Classes/pluginClass.m.tmpl

```diff
@@ -1,25 +1,20 @@
 #import "{{pluginClass}}.h"
 
-@implementation {{pluginClass}} {
+@implementation {{pluginClass}}
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
+  FlutterMethodChannel* channel = [FlutterMethodChannel
+      methodChannelWithName:@"{{projectName}}"
+            binaryMessenger:[registrar messenger]];
+  {{pluginClass}}* instance = [{{pluginClass}} new];
```

Comment:
I don't see a good reason to insist that instances can be created of all plugins? At least, we don't need that on the Java side.

The `instance` in question was supposed to have a life-span identical to the block that is given (via the MethodChannel) to the underlying `BinaryMessenger`---which means it should be kept alive for as long as the process runs. Is that not what happens?